sdk/src/wallet/account/operations/syncing/transactions.rs

```diff
@@ -89,11 +89,10 @@ impl Account {
             let TransactionEssence::Regular(essence) = transaction.payload.essence();
             let mut input_got_spent = false;
             for input in essence.inputs() {
-                if let Input::Utxo(input) = input {
-                    if let Some(input) = account_details.outputs.get(input.output_id()) {
-                        if input.is_spent {
-                            input_got_spent = true;
-                        }
+                let Input::Utxo(input) = input;
```

Comment:
Leave?

Reply:
I feel like these enums should use `#[non_exhaustive]` for this exact reason.

Reply:
Though I guess that wouldn't prevent this in the defining crate. It may be better to just do a match with only one variant and no `_` binding.

Reply:
I think clippy doesn't like when you do this

Reply:
Alright, fair enough I suppose